modules/Kconfig

```diff
@@ -30,3 +34,23 @@ source "modules/Kconfig.tinycbor"
 source "modules/Kconfig.tinycrypt"
 source "modules/Kconfig.vega"
 source "modules/Kconfig.xtensa"
+
+comment "Unavailable modules, please install those via the project manifest."
+
+# List of Zephyr modules text when module is not available, please use the
+# following syntax:
+# ---------------------------------------------------
+# comment "<module_name> module not available."
+#	depends on !ZEPHYR_<MODULE_NAME_UPPER>_MODULE
+#
+# Remember to add the following code inside the `<module>/Kconfig.<module> file:
+# ---------------------------------------------------
+# config ZEPHYR_<MODULE_NAME_UPPER>_MODULE
+# 	bool
+
+# This ensures that symbols are available in Kconfig for dependency checking
+# and referencing, while keeping the settings themselves unavailable when the
+# modules are not present in the workspace
+if 0
+osource "modules/*/Kconfig.*"
+endif
```

Comment:
This cannot be duplicated for the user's module roots AFAICT (without modifying this file). It might not be a big deal, just want to hear your thoughts.

Reply:
This is only for Zephyr natively handled Zephyr modules, and it is made like this to ensure that Kconfig symbols are always defined (but not active) when a module is not present.

This is in order to ensure that you do not get unknown Kconfig symbols when running Kconfig.

Like here, where there is a `depends on MBEDTLS` but that symbol would have been undefined without this osource, if `MBEDTLS` is not available:
https://github.com/zephyrproject-rtos/zephyr/blob/1c493b4a7706e9934949f8d23f8d8eddc7256200/subsys/random/Kconfig#L89

Users adding custom `MODULE_ROOT` may not have such problems in the first place.
That will solely depend on whether they consider modules in their downstream manifest to be optional.
And in case optional modules are allowed, that they too contain modules with such symbol dependencies.

Finally, it requires this to be a downstream fork of a downstream fork of Zephyr project, which is outside the scope of this PR.